packages/sources/cfbenchmarks-test/src/endpoint/common/crypto.ts

```diff
@@ -0,0 +1,127 @@
+import { InputParameters } from '@chainlink/external-adapter-framework/validation/input-params'
+import {
+  PriceEndpoint,
+  priceEndpointInputParameters,
+} from '@chainlink/external-adapter-framework/adapter'
+import { RoutingTransport } from '@chainlink/external-adapter-framework/transports/routing'
+import { EmptyObject } from '@chainlink/external-adapter-framework/util'
+import { AdapterInputError } from '@chainlink/external-adapter-framework/validation/error'
+import { transport as rest, transportSecondary as restSecondary } from '../rest/crypto'
+import {
+  transport as websocket,
+  transportSecondary as websocketSecondary,
+} from '../websocket/crypto'
+import { customSettings } from '../../config'
+
+export type Params = { index?: string; base?: string; quote?: string }
+
+const extendedInputParameters: InputParameters = {
+  index: {
+    description: 'The ID of the index. Takes priority over base/quote when provided.',
+    type: 'string',
+    required: false,
+  },
+  base: {
+    aliases: ['from', 'coin'],
+    type: 'string',
+    description: 'The symbol of symbols of the currency to query',
+    required: false,
+  },
+  quote: {
+    aliases: ['to', 'market'],
+    type: 'string',
+    description: 'The symbol of the currency to convert to',
+    required: false,
+  },
+}
+
+export const inputParameters = {
+  ...priceEndpointInputParameters,
+  ...extendedInputParameters,
+}
+
+export type EndpointTypes = {
+  Request: {
+    Params: Params
+  }
+  Response: {
+    Data: EmptyObject
+    Result: number
+  }
+  CustomSettings: typeof customSettings
+}
+
+const idOverrideFromBaseQuote: { [base: string]: { [quote: string]: string } } = {
+  BTC: {
+    USD: 'BRTI',
+  },
+}
+export const overrideId = (base: string, quote: string): string | undefined =>
+  idOverrideFromBaseQuote[base][quote]
+
+export const getPrimaryId = (base: string, quote: string): string => `${base}/${quote}_RTI`
+export const getSecondaryId = (base: string, quote: string): string => `U_${base}/${quote}_RTI`
+
+export const getIdFromBaseQuote = (
+  base: string,
+  quote: string,
+  type: 'primary' | 'secondary',
+): string => {
+  const override = overrideId(base, quote)
+  if (override) return override
+
+  if (type === 'secondary') return getSecondaryId(base, quote)
+  return getPrimaryId(base, quote)
+}
```

Comment:
We could maybe move all these helper fns to a separate file

Reply:
Sure. I believe they are specific to the use of this crypto endpoint. But may be cleaner.